### PR TITLE
Agg setup

### PR DESCRIPTION
**Pull request by bjlittle**
This PR brings the agg source into the repo for the sole purpose of building `agg-regrid` in a self-contained way (akin to `matplotlib`).

This will allow `agg-regrid` to now build in a conda recipe.

### CODE REVIEW

File: INSTALL.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+============
+
+To build and install agg-regrid from source::
+
+    python setup.py build_ext
+    python setup.py install
+
+
+Dependencies
+------------
```
Comment (pelson) [inline INSTALL.rst]: Missing: numpy, cython, iris

**Comment by pelson** [comment]
Turns out this is actually failing to build on my Mac.

```
(agg_regrid) pelson@~/dev/scitools/iris-agg-regrid> pip install -e .
Obtaining file:///Users/pelson/dev/scitools/iris-agg-regrid
Installing collected packages: agg-regrid
  Running setup.py develop for agg-regrid
    Complete output from command /Users/pelson/miniconda/bin/python -c "import setuptools, tokenize;__file__='/Users/pelson/dev/scitools/iris-agg-regrid/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" develop --no-deps:
    running develop
    running egg_info
    writing dependency_links to agg_regrid.egg-info/dependency_links.txt
    writing agg_regrid.egg-info/PKG-INFO
    writing top-level names to agg_regrid.egg-info/top_level.txt
    reading manifest file 'agg_regrid.egg-info/SOURCES.txt'
    writing manifest file 'agg_regrid.egg-info/SOURCES.txt'
    running build_ext
    building 'agg_regrid._agg' extension
    creating build
    creating build/temp.macosx-10.9-x86_64-3.5
    creating build/temp.macosx-10.9-x86_64-3.5/Users
    creating build/temp.macosx-10.9-x86_64-3.5/Users/pelson
    creating build/temp.macosx-10.9-x86_64-3.5/Users/pelson/dev
    creating build/temp.macosx-10.9-x86_64-3.5/Users/pelson/dev/scitools
    creating build/temp.macosx-10.9-x86_64-3.5/Users/pelson/dev/scitools/iris-agg-regrid
    creating build/temp.macosx-10.9-x86_64-3.5/Users/pelson/dev/scitools/iris-agg-regrid/agg_regrid
    gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/pelson/miniconda/include -arch x86_64 -Iagg_regrid -I/Users/pelson/dev/scitools/iris-agg-regrid/agg_regrid -I/Users/pelson/dev/scitools/iris-agg-regrid/extern/agg-2.4/include -I/Users/pelson/miniconda/lib/python3.5/site-packages/numpy/core/include -I/Users/pelson/miniconda/include/python3.5m -c /Users/pelson/dev/scitools/iris-agg-regrid/agg_regrid/_agg.cpp -o build/temp.macosx-10.9-x86_64-3.5/Users/pelson/dev/scitools/iris-agg-regrid/agg_regrid/_agg.o
    In file included from /Users/pelson/dev/scitools/iris-agg-regrid/agg_regrid/_agg.cpp:564:
    In file included from /Users/pelson/miniconda/lib/python3.5/site-packages/numpy/core/include/numpy/arrayobject.h:4:
    In file included from /Users/pelson/miniconda/lib/python3.5/site-packages/numpy/core/include/numpy/ndarrayobject.h:18:
    In file included from /Users/pelson/miniconda/lib/python3.5/site-packages/numpy/core/include/numpy/ndarraytypes.h:1777:
    /Users/pelson/miniconda/lib/python3.5/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: "Using deprecated NumPy API, disable it by "          "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-W#warnings]
    #warning "Using deprecated NumPy API, disable it by " \
     ^
    1 warning generated.
    gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/pelson/miniconda/include -arch x86_64 -Iagg_regrid -I/Users/pelson/dev/scitools/iris-agg-regrid/agg_regrid -I/Users/pelson/dev/scitools/iris-agg-regrid/extern/agg-2.4/include -I/Users/pelson/miniconda/lib/python3.5/site-packages/numpy/core/include -I/Users/pelson/miniconda/include/python3.5m -c /Users/pelson/dev/scitools/iris-agg-regrid/agg_regrid/_agg_raster.cpp -o build/temp.macosx-10.9-x86_64-3.5/Users/pelson/dev/scitools/iris-agg-regrid/agg_regrid/_agg_raster.o
    In file included from /Users/pelson/dev/scitools/iris-agg-regrid/agg_regrid/_agg_raster.cpp:29:
    /Users/pelson/dev/scitools/iris-agg-regrid/extern/agg-2.4/include/agg_scanline_u.h:467:30: error: type 'base_type' (aka 'agg::scanline_u8') is not a direct or virtual base of 'scanline32_u8_am<AlphaMask>'
            scanline32_u8_am() : base_type(), m_alpha_mask(0) {}
                                 ^~~~~~~~~
    /Users/pelson/dev/scitools/iris-agg-regrid/extern/agg-2.4/include/agg_scanline_u.h:468:49: error: type 'base_type' (aka 'agg::scanline_u8') is not a direct or virtual base of 'scanline32_u8_am<AlphaMask>'
            scanline32_u8_am(const AlphaMask& am) : base_type(), m_alpha_mask(&am) {}
                                                    ^~~~~~~~~
    /Users/pelson/dev/scitools/iris-agg-regrid/extern/agg-2.4/include/agg_scanline_u.h:473:24: error: call to non-static member function without an object argument
                base_type::finalize(span_y);
                ~~~~~~~~~~~^~~~~~~~
    /Users/pelson/dev/scitools/iris-agg-regrid/extern/agg-2.4/include/agg_scanline_u.h:476:64: error: call to non-static member function without an object argument
                    typename base_type::iterator span = base_type::begin();
                                                        ~~~~~~~~~~~^~~~~
    /Users/pelson/dev/scitools/iris-agg-regrid/extern/agg-2.4/include/agg_scanline_u.h:477:45: error: call to non-static member function without an object argument
                    unsigned count = base_type::num_spans();
                                     ~~~~~~~~~~~^~~~~~~~~
    /Users/pelson/dev/scitools/iris-agg-regrid/extern/agg-2.4/include/agg_scanline_u.h:481:60: error: call to non-static member function without an object argument
                                                    base_type::y(),
                                                    ~~~~~~~~~~~^
    6 errors generated.
    error: command 'gcc' failed with exit status 1
    
    ----------------------------------------
Command "/Users/pelson/miniconda/bin/python -c "import setuptools, tokenize;__file__='/Users/pelson/dev/scitools/iris-agg-regrid/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" develop --no-deps" failed with error code 1 in /Users/pelson/dev/scitools/iris-agg-regrid/
```

That is with:

```
(agg_regrid) pelson@~/dev/scitools/iris-agg-regrid> g++ -v
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 9.0.0 (clang-900.0.39.2)
Target: x86_64-apple-darwin17.2.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

**Comment by pelson** [comment]
We should disable the use of the deprecated Numpy API to prevent those warnings too.

**Comment by pelson** [comment]
Despite all of the above (which I think we need to deal with), I'm still going to merge and move forwards.

**Comment by bjlittle** [comment]
@pelson Much appreciated, thanks. 

Also, do you ever sleep?

**Comment by pelson** [comment]
Might say the same thing about you with the time of this PR... 😴 🛌

**Pull request merged**